tests/integration/customizations/s3/__init__.py

```diff
@@ -25,9 +26,16 @@ def make_s3_files(session, key1='text1.txt', key2='text2.txt'):
     """
     region = 'us-west-2'
     bucket = create_bucket(session)
-    string1 = "This is a test."
-    string2 = "This is another test."
-    client = session.create_client('s3', region_name=region)
+
+    if size:
+        string1 = "*" * size
+        string2 = string1
+    else:
+        string1 = "This is a test."
+        string2 = "This is another test."
+
+    config = Config(s3={"addressing_style": "path"})
```

Comment:
The issue is boto/botocore#760. The issue is that we're forcing a large packet to an endpoint that is trying to tell us to re-direct. The service closes the connection since we continue to send data. This doesn't occur with path style since there are no DNS propagation requirements, and doesn't occur with small payloads because the service will just forward on small packets.